src/Integration.Vsix/CFamily/CLangAnalyzer.cs

```diff
@@ -110,6 +119,17 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
         {
             Debug.Assert(IsAnalysisSupported(detectedLanguages));
 
+            var request = await TryCreateRequestAsync(path, analyzerOptions);
+
+            if (request != null)
+            {
+                await RunAnalysisAsync(request, consumer, statusNotifier, cancellationToken);
+            }
+        }
+
+        protected virtual /* for testing */ async Task<IRequest> TryCreateRequestAsync(string path, IAnalyzerOptions analyzerOptions)
```

Comment:
This new method is virtual so the testable subclass can override it and provide a synchronisation method so tests running on the "main" thread can wait until this method (on another thread) has finished.